### ISSUE TITLE
Electrum 0.3 crashes

### ISSUE THREAD

**Issue by retroverseworld** [spesmilo/electrum#7893]
```
$ python electrum.py create
Password (hit return if you do not wish to encrypt your wallet):
in order to use wallet encryption, please install pycrypto  (sudo easy_install pycrypto)
server (default:ecdsa.org):
port (default:50000):
fee (default 0.005):
if you are restoring an existing wallet, enter the seed. otherwise just press enter:
Your seed is 12f2458bab0372ecc4d2097f0df39167
Please store it safely
Traceback (most recent call last):
  File "electrum.py", line 715, in <module>
    wallet.create_new_address(False, None)
  File "electrum.py", line 295, in create_new_address
    secexp = ecdsa.util.randrange_from_seed__trytryagain( seed, order )
  File "/usr/local/lib/python2.7/dist-packages/ecdsa/util.py", line 193, in randrange_from_seed__trytryagain
    extrabyte = int2byte(ord(generate(1)) & lsb_of_ones(extrabits))
  File "/usr/local/lib/python2.7/dist-packages/ecdsa/util.py", line 98, in __call__
    a = [next(self.generator) for i in range(numbytes)]
  File "/usr/local/lib/python2.7/dist-packages/ecdsa/util.py", line 109, in block_generator
    ("prng-%d-%s" % (counter, seed)).encode()
UnicodeDecodeError: 'ascii' codec can't decode byte 0xb6 in position 7: ordinal not in range(128)
```
Trying to restore seed from 2011; which does not work with later Electrum versions as  wallet recovered in later Electrum versions show zero BTC balance.

**Comment by SomberNight** [comment]
What do you mean by "Electrum 0.3"?

Every seed created in Electrum 0.34 (released around 2011-12-16) or later is still supported in modern versions.
Electrum 0.25 was the first public version, [released 2011-11-05](https://bitcointalk.org/index.php?topic=50936.msg607510#msg607510).

See https://github.com/spesmilo/electrum/issues/6844#issuecomment-743729476

**Comment by retroverseworld** [comment]
This https://github.com/spesmilo/electrum/tree/5883aaf8ca2f79bf694d11ac6b63f5defd2a2c38 Electrum 0.30 is the version I've tried. The next earliest version I can find is Electrum 0.56. That fails to restore the wallet completely or to create a new wallet completely as it's unable to connect to any server or the Bitcoin blockchain I guess.  It just says Please wait Addresses generated: 0, Kilobytes received 0.0. Because I created my wallet all the way back in about 2011 with approx 800-1500BTC, it looks like I may not be able to restore my wallet from my seed. Is there any chance you can update the code to allow it to connect some how. 
So; the version of Electrum I used would have permitted arbitrary words. One word from my seed that it generated is 'cranky'. So from this information are you able to work out what version of Electrum would have allowed that.  Another word is 'absurd'. Please help; there is a lot to ride on this. These words are not in mnemonic.py but there were definitely generated by Electrum back in 2011 or  slightly earlier. I don't remember the exact dates. Was there ever a beta version of Electrum on the internet.

**Comment by SomberNight** [comment]
> So; the version of Electrum I used would have permitted arbitrary words. One word from my seed that it generated is 'cranky'. So from this information are you able to work out what version of Electrum would have allowed that. Another word is 'absurd'. 

Versions prior to 2.0 used a fixed wordlist. Seedphrases could only contain words from that list.

**Comment by retroverseworld** [comment]
Thank you SomberNight. Can you help me answer these questions please:
1) Why is it my 12 word seed from 2011  is accepted in all versions of Electrum that I've used to restore my wallet since 2016.  The only problem is that the restored wallets display a balance of zero BTC and no history of any transactions; when I know that there were transactions and a balance of east 800 BTC. If I alter even one word from the seed, Electrum does not accept the seed so that the wallet cannot be restored.
2) You say Electrum versions prior to 2.0 used a fixed wordlist. Does that mean after 2.0 Electrum used any random 12 word seed.
3) If it wasn't Electrum then what else could it have been. I know the wallet I used had a GUI  that looked just like Electrum's and permitted me to send coins between different wallets from. I did this successfully just to be sure I understood the process of sending and receiving coins because it was quite a novel concept to me. I had to install the wallet on my windows XP PCs and laptop. I do not think there were any online exchange wallets back then. 
Thank you.

**Comment by SomberNight** [comment]
> 1. Why is it my 12 word seed from 2011 is accepted in all versions of Electrum that I've used to restore my wallet [...]
> If I alter even one word from the seed, Electrum does not accept the seed so that the wallet cannot be restored.

What type of seed do you have? You can see this when you enter the seed in the program, just below the text box. ("Seed Type")
![asd](https://user-images.githubusercontent.com/29142493/179096282-d02e9a65-3b6e-4f9c-9568-cdc34f295b26.png)

> 2. You say Electrum versions prior to 2.0 used a fixed wordlist. Does that mean after 2.0 Electrum used any random 12 word seed.

No. The hash of the seedphrase text needs to satisfy certain constraints, however there is no wordlist involved. As a consequence, arbitrary text is usually not a valid seed, however any garbled text *might* be valid. E.g. `9dk` happens to be a valid seed.
See https://electrum.readthedocs.io/en/latest/seedphrase.html

> 3. If it wasn't Electrum then what else could it have been [...] I had to install the wallet on my windows XP PCs and laptop.

Like I have already said:

>> See https://github.com/spesmilo/electrum/issues/6844#issuecomment-743729476
>
> Version 0.34 was released around ~40 days after Electrum became publicly available at all, in late 2011. At that time, there were no binaries, you could only run from source. It was only available on the bitcointalk forums, and you would have had to closely follow the forums to even notice it existed. Bugs were all too common, and there was a release every few days.
>
> You would know if you had such a seed.

As above, in 2011, there were no Windows binaries for these very early versions.

**Comment by retroverseworld** [comment]
Thank you once more.  Clearly  many things don't add up here.  Maybe my memory deceives me.
1. Tried restoring wallet with Electrum 4.1.5 and seed type is standard, i.e. Electrum type seed. I only see segwit if I first select option BIP39, then native segwit. 
![image](https://user-images.githubusercontent.com/97367943/179315950-163793a5-742c-4eb6-b69f-bb60623fdcf9.png)

2. Just in case I have my timelines wrong, what was the first desktop  Electrum Version available for windows.
3. I had almost given up hope when I found another old set of phrases from which I successfully recovered a wallet with historical transactions. I'm not 100% sure that the wallet belongs to me although I'm 100% sure the seed is mine as it's written in my notebook in my own hand writing. One of the addresses from that restored seed is: 17HF2zJE5SWezC5VJF3uXfve3CuxPdDAUA	       60.25105
This makes me believe my original seed was generated by Electrum; a binary version because I would not have been able to mess around with python, I had not even heard of python programming language back in 2011 or even a few years post that.
From the link you provided it sounds like whether my seed phrase was generated pre Electrum 2.0 or post it should work with later versions of Electrum including 4.1.5.  So when I recover my old wallet with my seed, it should show the correct balance of my BTC and historical transactions. Yet it does not.
So either my seed phrase is wrong somehow; or my seed phrase is somehow incompatible with current versions of Electrum; or my seed was not generated by Electrum. 
I don't really know what to do now. Maybe one day I will donate my seed to science and see if some PHD student can solve the mystery. More than 7 yrs of searching for an answer and I'm nowhere near finding it. I'm trumped.

**Comment by SomberNight** [comment]
> Tried restoring wallet with Electrum 4.1.5 and seed type is standard

"standard" means the seed was generated using Electrum 2.0 or later.
Seed generated before 2.0 have type "old".

Is it this "standard" seed that contains the word "cranky"? I wonder how that word could end up in the seed...

> what was the first desktop Electrum Version available for windows.

Not sure... I expect a few months after the first release, so some time in 2012 there would have been binaries, but I don't know exactly when. You would have to read old forum archives to find out.

**Comment by retroverseworld** [comment]
I apologise that my memory does not serve me well; but I do have firm memories of:
1) Purchasing the bitcoin to the tune of 800 to 1500 coins.
2) The web sight instructed to transfer the coins to a wallet.
3) The wallet instructed to memorise the seed phrase of 12 words and not write them down but I wrote them down any way.
4) I played around with the wallet transferring small amounts of bitcoin between two or three wallets; possibly same wallet but different addresses. I remember there being a contact tab in the wallet because I remember thinking that was a useful feature at the time.
5) I'm convinced this wallet was Electrum; but I could be wrong.
Someone in bitcoin forum mentioned bitaddress.org. It looked remarkably familiar and also uses seeds. I'll try to figure out, how to use bitaddress to recover addressses from my 12 word and see if I can locate the addresses on blockchain explorer. 
I would like to keep this issue open so I can provide an update or in case I've further questions; otherwise thank you so much for all your answers.
